lib/cib/cib_remote.c

```diff
@@ -210,9 +211,34 @@ cib_remote_callback_dispatch(gpointer user_data)
 
     xmlNode *msg = NULL;
 
-    crm_info("Message on callback channel");
+    /* If start_time is 0, we've previously handled a complete message and this
+     * callback is being reused for a new message.  Reset the start_time, giving
+     * this new message timeout_sec from now to complete.
+     */
+    if (rd->start_time == 0) {
+        rd->start_time = time(NULL);
+    } else if (time(NULL) >= rd->start_time + rd->timeout_sec) {
+        crm_info("Error reading from remote client: %s", pcmk_rc_str(ETIME));
+        return -1;
+    }
 
-    rc = pcmk__read_remote_message(&private->callback, rd->timeout_sec);
+    rc = pcmk__read_available_remote_data(&private->callback);
+    switch (rc) {
+        case pcmk_rc_ok:
+            /* We have the whole message so process it */
+            break;
+
+        case EAGAIN:
+            /* We haven't read the whole message yet */
+            return 0;
+
+        default:
+            /* Error */
+            crm_info("Error reading from remote client: %s", pcmk_rc_str(rc));
+            return -1;
+    }
+
+    crm_info("Message on callback channel");
```

Comment:
probably not worth a log after you're done debugging

could be worth an xml trace though, in `pcmk__remote_message_xml()` or after calling it